v2/gcs-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/utils/GCSReader.java

```diff
@@ -204,4 +218,49 @@ private List<TrimmedShardedDataChangeRecord> checkAndReturnIfFileExists() {
           " Cannot determine file creation progress for shard : " + shardId, e);
     }
   }
+
+  private List<TrimmedShardedDataChangeRecord> waitTillFileCreatedAndReturn() {
+    boolean found = false;
+    List<TrimmedShardedDataChangeRecord> changeStreamList = new ArrayList<>();
+    while (!found) {
+      try (InputStream stream =
```

Comment:
Existing code from getRecords() can be refactored and reused here

Reply:
Could not find a good way to do that. But have opened a general bug for code-refactoring during integration testing, will handle there.